### PR TITLE
add support for permissioned consume events

### PR DESCRIPTION
**Pull request by badass-commits**

### CODE REVIEW

**Comment by armaniferrante** [comment]
Also need to add a change to initializeMarket. See https://github.com/project-serum/serum-ts/pull/167/files#diff-402abe72f0ad3f1d14144de811e6dff0af20cc20c6caa8c0db42e07245fffc41R164

**Comment by badass-commits** [comment]
> Also need to add a change to initializeMarket. See https://github.com/project-serum/serum-ts/pull/167/files#diff-402abe72f0ad3f1d14144de811e6dff0af20cc20c6caa8c0db42e07245fffc41R164

thanks, added

File: packages/serum/src/market-proxy/middleware.ts [hunk 1]
```diff
@@ -13,6 +13,8 @@ export interface Middleware {
   settleFunds(ix: TransactionInstruction): void;
   closeOpenOrders(ix: TransactionInstruction): void;
   prune(ix: TransactionInstruction): void;
+  consumeEvents(ix: TransactionInstruction): void;
```
Comment (armaniferrante) [inline packages/serum/src/market-proxy/middleware.ts:16]: Want to do this in a separate PR? Since this should correspond with changes to https://github.com/project-serum/anchor/blob/master/spl/src/dex/middleware.rs#L44 .

Comment (armaniferrante) [reply packages/serum/src/market-proxy/middleware.ts:16]: Or alternatively make the corresponding rust changes.

Comment (badass-commits) [reply packages/serum/src/market-proxy/middleware.ts:16]: I made rust changes. https://github.com/project-serum/anchor/pull/838
